File: 02_stack/python/stack.py
class Stack () : 

    def __init__(self, capacity=5):
        self.capacity = capacity
        self.items = [None] * capacity 
        self.top = -1 

    def isFull (self) : 
        return self.top == self.capacity -1  
    def isEmpty (self) : 
        return self.top == -1 
    def push (self, val): 
        if self.isFull() : 
            return "Stack is full.!"
        else :
            self.top  +=1 
            self.items[self.top] = val 
            return f"{val} is added successfully!"
    def pop (self) : 
        if self.isEmpty() : 
            return "Stack is UnderFlow"
        else : 
            val = self.items[self.top]
            self.top -=1
            return val 
    def getTop (self): 
        if self.isEmpty(): 
            return "Stack is Empty"
        return self.items[self.top]

    def getStack (self) : 
        return self.items

myStack = Stack() 

print ("-----------ISFULL Function------------", end='\n')
print(myStack.isFull())
print ("-----------ISEMPTY Function------------", end='\n')
print(myStack.isEmpty())
print ("-----------PUSH 5 Numbers------------", end='\n')
print(myStack.push(10))
print(myStack.push(20))
print(myStack.push(30))
print(myStack.push(40))
print(myStack.push(50))
print ("-----------Print the Stack ------------", end='\n')
print(myStack.getStack())
print ("-----------Add values or items more than capacity------------", end='\n')
print(myStack.push(60))
print ("-----------check if the more value added or no ------------", end='\n')
print(myStack.getStack())
print ("-----------Remove Last index  Function------------", end='\n')
print(myStack.pop(50))
print ("-----------Check last index deletion------------", end='\n')
print(myStack.getTop())

File: 02_stack/cpp/stack.cpp
#include <iostream>
using namespace std ; 
const int MAX_SIZE = 100 ; 

class Stack {
    int top ; 
    int item[MAX_SIZE]; 
   
public : 
    Stack() : top(-1){} 

    bool isEmpty (){
        return top == -1 ;
    }

    bool isFull(){
        return top == MAX_SIZE -1 ; 
    }

    void push (int val){
        if(isFull()){
           cout << "Stack is Overwhelmed"; 
        }
        else {
            top ++ ; 
            item[top] = val ; 
        }
    }

    void pop (int&val){
        if (isEmpty()){
            cout << "Stack is Undeflow" ; 
        }
        else {
            val = item[top]; 
            top -- ; 
        }
    }

    void getTop () {
      cout << item[top] ; 
    }

    void getStack () {
       cout <<"Stack elements : ["; 
       for (int i=0; i <=top; i++){
           cout << item[i] << "," ; 
       }
       cout << "]\n" ; 
    }
}; 

int main() {
    Stack myStack;
    
    cout << "=== PUSHING ELEMENTS ===" << endl;
    myStack.push(10);
    myStack.push(20);
    myStack.push(30);
    
    cout << "\n=== STATUS ===" << endl;
    myStack.getStack();
    myStack.getTop();
    
    cout << "\n=== POPPING ELEMENT ===" << endl;
    int poppedVal;
    myStack.pop(poppedVal);
    cout << "-> Popped Value: " << poppedVal << "\n";
    
    cout << "\n=== FINAL STATUS ===" << endl;
    myStack.getStack();
    myStack.getTop();
    return 0;
}